Support for multiple checksums

johnbradley opened the pull request:
Adds support to check all known hashes attached to a file when downloading. Before this change DukeDSClient only checked the first MD5 hashes found. DukeDSClient will now validate all MD5 hashes. Files that have both valid and invalid hashes will have status CONFLICTED. If there are CONFLICTED files a warning will be printed but no error raised. If there are any FAILED hashes an error will be raised.

Example terminal output after downloading a project with a conflicted file:
```
...
Verifying contents of 2 downloaded files using file hashes.
mh/me.jpeg 0106a7dcf3ab09bc47cab34a7ff1fbba md5 OK
mh/stuff.txt 7e11867261e621458a06f3e6bc301c8b md5 CONFLICTED
All downloaded files have at least one valid hash.

WARNING: Some downloaded files also have invalid hashes.

```

Example project in DukeDS dev that has multiple checksums: https://dev.dataservice.duke.edu/#/project/cfa04bfd-c025-4025-8153-b058d0893a12
The file in the above project has an additional wrong checksum added via the [PUT /api/v1/uploads/{id}/hashes](
https://apidev.dataservice.duke.edu/apiexplorer#!/uploads/putApiV1UploadsIdHashes) DukeDS API.


## Background
There was a bug in the DukeDS web portal where [large files uploaded had an incorrect hash]( https://github.com/Duke-Translational-Bioinformatics/duke-data-service-portal/issues/901).
A project was created to [add the correct hash to these files](https://github.com/Duke-Translational-Bioinformatics/md5_reporter). This leaves the incorrect hash in the database for these files.

Fixes #256

johnbradley commented:
Any feedback on the verbiage would be appreciated.

dleehr commented on ddsc/core/download.py:
```diff
@@ -544,32 +549,94 @@ def __init__(self, algorithm, expected_hash_value, file_path):
+
+class FileHashStatus(object):
+    STATUS_OK = "OK"
+    STATUS_CONFLICTED = "CONFLICTED"
```
The word `CONFLICTED` to me suggests there's a situation that needs to be resolved. Since we're not recommending any action, I would choose a different word. Specifically in the printed list of statuses after download/verification,`CONFLICTED` makes me think there's a problem with the file and I should re-download or something. I'd suggest changing this to something else like `WARNING` or `OK (1 of 3 matched)`.

johnbradley commented:
Changed CONFLICTED to WARNING.
Updated command line when downloading:
```
$ ddsclient download -p MultiHashes mh
Fetching list of files/folders.
Downloading 2 files.
Done: 100%                          
Verifying contents of 2 downloaded files using file hashes.
mh/me.jpeg 0106a7dcf3ab09bc47cab34a7ff1fbba md5 OK
mh/stuff.txt 7e11867261e621458a06f3e6bc301c8b md5 WARNING
All downloaded files have been verified successfully.

NOTICE: Data Service reports multiple hashes for 1 files. 
The downloaded files have been verified and confirmed to match one of these hashes. 
You do not need to retry the download. 
For more information, visit https://github.com/Duke-GCB/DukeDSClient/wiki/MD5-Hash-Conflicts.

```

dleehr commented on ddsc/core/download.py:
```diff
@@ -14,6 +14,12 @@
 SWIFT_EXPIRED_STATUS_CODE = 401
 S3_EXPIRED_STATUS_CODE = 403
+MISMATCHED_FILE_HASH_WARNING = """
+NOTICE: Data Service reports multiple hashes for {} files.
```
Would be nice to optionally pluralize `file`
